src/gui/src/mainWindow.cpp

```diff
@@ -285,6 +286,48 @@ void MainWindow::createToolbars()
   view_tool_bar_->setObjectName("view_toolbar");  // for settings
 }
 
+const std::string MainWindow::addToolbarButton(const std::string& name,
+                                               const QString& text,
+                                               const QString& script,
+                                               bool echo)
+{
+  auto action = view_tool_bar_->addAction(text);
+
+  connect(action, &QAction::triggered, [script, echo, this]() {
+    script_->executeCommand(script, echo);
+  });
+
+  // ensure key is unique
+  std::string key = name;
+  int key_idx = 0;
+  while (buttons_.count(key) != 0) {
+    key = name + std::to_string(key_idx);
+    key_idx++;
+  }
```

Comment:
I think it is better to error than rename.   It is a bit non-intuitive to return a button with a different name and may cause confusion when using remove button.

Reply:
you are correct that it is less intuitive.
Maybe the right thing to do would be:
1. If name is defined, use it (and return the name) or throw an error
2. If name is empty, generate a new name and return that name

What do you think?

Reply:
I went ahead and changed it to the behavior above.